internal/platform/platform.go

```diff
@@ -24,13 +31,55 @@ func (hp *HardwarePlatform) PciDevices() ([]*ghw.PCIDevice, error) {
 	return pciInfo.Devices, nil
 }
 
+func (hp *HardwarePlatform) NetDevs() ([]*ghw.NIC, error) {
+	netInfo, err := ghw.Network()
+	if err != nil {
+		return nil, err
+	}
+	return netInfo.NICs, nil
+}
+
 func (hp *HardwarePlatform) Product() (*ghw.ProductInfo, error) {
 	return ghw.Product()
 }
 
+func (hp *HardwarePlatform) ReadDeviceSerialNumber(pciDevice *ghw.PCIDevice) (string, error) {
```

Comment:
nicely done. Any reason to pass in a pointer? Yes it costs a copy in a garbage collected language, but the price is that I don't know for sure if pciDevice is only an input or also an output.

Reply:
Personal preference. In C++ it would be static const pointer but we don't have that in golang.... 

I like pointers rather than copies. Would a function commentary suffice?

Reply:
Should be ok... Maybe golang will support const pointers which would make alot of sense from a programming language point of view.